### PR TITLE
docs: Updating README with repository details.

### PR DESCRIPTION
**Pull request by davidjoy**
This includes both the component's dependency on @edx/frontend-platform, as well as its required environment variables.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,61 +1,95 @@
...
 
-frontend-component-footer is a library containing a site footer
-component for use when building Open edX frontend applications.
+********
+Overview
+********
 
-Usage
------
+A generic footer for Open edX micro-frontend applications.  It includes a logo with trademark, and an optional language selector dropdown.
```
Comment (sarina) [inline README.rst:11]: what does "a logo with trademark" mean?

File: README.rst [hunk 2]
```diff
@@ -1,61 +1,95 @@
...
+  <Footer
+    onLanguageSelected={(languageCode) => {/* set language */}}
+    supportedLanguages={[
+      { label: 'English', value: 'en'},
+      { label: 'Español', value: 'es' },
+    ]}
+  />
+
+* `An example of minimal component and messages usage. <https://github.com/edx/frontend-template-application/blob/master/src/index.jsx#L23>`_
+* `An example of SCSS file usage. <https://github.com/edx/frontend-template-application/blob/master/src/index.scss#L9>`_
```
Comment (sarina) [inline README.rst:71]: here too have the concern about linking to master

File: README.rst (hunk 1 above)
Comment (davidjoy) [reply README.rst:11]: I expanded the description - good thought.

File: README.rst (hunk 2 above)
Comment (davidjoy) [reply README.rst]: Fixed!

**Review by sarina — APPROVED** [review]

**Pull request merged**